src/pokemon-service/utils/file.go

```diff
@@ -62,3 +67,60 @@ func (f file) AppendLineToFile(line string) error {
 
 	return nil
 }
+
+func (f file) ReadAllFileConcurrent(even bool, items, items_per_worker int) ([]string, error) {
```

Comment:
Here is everything that the reviewer needs to know on the concurrency, everything else in the pr is to allow to the new endpoint to reach the concurrency reading